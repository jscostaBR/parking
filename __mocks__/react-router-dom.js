/* eslint-disable no-undef */
const originalModule = jest.requireActual('react-router-dom')
module.exports = {
  __esModule: true,
  ...originalModule,
  useLocation: jest.fn().mockReturnValue({ pathname: '/'}),
  useNavigate: jest.fn()
}

