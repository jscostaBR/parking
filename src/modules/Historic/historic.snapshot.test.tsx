import * as React from 'react'
import { Historic } from '.'
import renderer from 'react-test-renderer'
import { HashRouter } from 'react-router-dom'

describe('render Historic', () => {

  jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
    useNavigate: () => ({ url: '/' }),
  }));

  it('render correctly', () => {
    const tree = renderer.create(
      <HashRouter>
        <Historic plate="AAA-1234" />
      </HashRouter>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
