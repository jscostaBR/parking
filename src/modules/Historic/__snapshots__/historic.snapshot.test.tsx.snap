// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Historic render correctly 1`] = `
[
  <nav
    className="sc-dkzDqf kIIFRG"
  >
    <img />
    <img
      className="hamburger"
      onClick={[Function]}
    />
  </nav>,
  <div
    className="sc-hKMtZM kVBDyd"
  >
    <div
      className="header"
    >
      <img
        className="logo"
      />
      <img
        className="close"
        onClick={[Function]}
      />
    </div>
    <div>
      <span>
        Entrada
      </span>
      <span>
        Sa√≠da
      </span>
    </div>
  </div>,
  <div
    className="sc-bczRLJ jLxbrI"
  >
    <section>
      <div
        className="header"
      >
        <img
          onClick={[Function]}
        />
        Placa 
        AAA-1234
      </div>
    </section>
  </div>,
]
`;
